xlsynth-driver/src/dslx_equiv.rs

```diff
@@ -62,22 +70,91 @@ fn build_ir_for_dslx(
+        // again here to avoid duplicating logic deep inside the converter path;
+        // this is still a single path in this function, vs. re-parsing
+        // elsewhere).
+        let domains = if want_domains {
+            // Parse/typecheck via runtime API
+            use xlsynth::dslx;
+            let mut import_data = dslx::ImportData::new(dslx_stdlib_path, &additional_search_paths);
+            let tcm = dslx::parse_and_typecheck(
+                &dslx_contents,
+                input_path.to_str().unwrap(),
+                module_name,
+                &mut import_data,
+            )
+            .expect("parse_and_typecheck success");
+            let module = tcm.get_module();
+
+            let type_info = tcm.get_type_info();
+            let mut domains: ParamDomains = HashMap::new();
+            for i in 0..module.get_member_count() {
+                if let Some(dslx::MatchableModuleMember::Function(f)) =
+                    module.get_member(i).to_matchable()
+                {
+                    if f.get_identifier() == dslx_top {
+                        for pidx in 0..f.get_param_count() {
+                            let p = f.get_param(pidx);
+                            let name = p.get_name();
+                            let ta = p.get_type_annotation();
+                            let ty = type_info.get_type_for_type_annotation(&ta);
+                            if ty.is_enum() {
+                                let enum_def = ty.get_enum_def().unwrap();
+                                let mut values = Vec::new();
+                                for mi in 0..enum_def.get_member_count() {
+                                    let m = enum_def.get_member(mi);
+                                    let expr = m.get_value();
+                                    // Use the owner module's type info for the enum value
+                                    // expression.
+                                    let owner_module = expr.get_owner_module();
+                                    let owner_type_info = tcm
+                                        .get_type_info_for_module(&owner_module)
+                                        .expect("imported type info");
+
+                                    // let owner_type_info = if owner_module == module {
+                                    //     &type_info
+                                    // } else {
+                                    //     &type_info.get_imported_type_info(&owner_module).expect("
+                                    // imported type info") };
+
+                                    // let owner_type_info = tcm
+                                    //     .get_type_info_for_module(&owner_module)
+                                    //     .expect("owner module type info");
+                                    let interp =
+                                        owner_type_info.get_const_expr(&expr).expect("constexpr");
```

Comment:
we should probaly have a multi-module unit test that shows why this "owner is not where the enum is defined" test case